### PR TITLE
Add config option to build netCDF tools

### PR DESCRIPTION
**Pull request by derobins**
* Adds --enable-netcdf-tools to Autotools
* Adds HDF4_BUILD_NETCDF_TOOLS to CMake

Both configure whether or not the netCDF tools ncdump and ncgen will be built.

As a part of this change, the HDF4 tools were set to be built by default (they were previously NOT built by default). This also updates the --enable-netcdf help text and libhdf4.settings file.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -202,13 +202,21 @@ AC_SUBST(TEST_FORTRAN_NETCDF) TEST_FORTRAN_NETCDF="yes"
...
-                              [Build HDF4 versions of NetCDF APIs (version 2.3.2) [default=yes]])],,
+                              [HDF4 builds replacement netCDF 2.3.2 API calls that are exported (e.g., ncopen()).
+                               Enabling this option uses the original netCDF names when building these functions.
+                               Disable prefixes them with 'sd_' to avoid name clashes.
+                               This does NOT affect building the HDF4-built netCDF tools ncdump and ncgen,
+                               nor does it disable any HDF4 functionality - this option only affects
+                               API call names.
+                               Disabling this option disables building the netCDF Fortran wrappers (they
+                               will not be built with either the original or sd_ prefixed names).
+                               [default=yes (HDF4-built netCDF API calls will be exported undecorated)]])],,
```
Comment (derobins) [inline configure.ac:213]: This is such a confusing option. After 4.2.16 releases, I'll do some internal renaming to make the purpose of this option clearer to maintainers.

File: release_notes/RELEASE.txt [hunk 2]
```diff
@@ -38,6 +38,33 @@ New features and changes
...
     Configuration:
     -------------
+    - Tools are now built by default in CMake
+
+      The option to build and test the HDF4 tools hdiff, hrepack, hdfimport,
+      and dumper was disabled by default. It is now enabled by default.
+
+      This option also controls the ability to build the netCDF tools.
+
+      (DER - 2023/02/28)
```
Comment (derobins) [inline release_notes/RELEASE.txt:48]: Unclear why this was the default, but CMake should do what the Autotools do.

**Comment by derobins** [comment]
Autotools / CMake parity in HDF4 is poor :(

File: configure.ac (hunk 1 above)
Comment (derobins) [reply configure.ac:213]: It's not really "enabling" functionality, it's just controlling name-mangling, so "enable netCDF" is a terrible name.

**Review by byrnHDF — APPROVED** [review]

File: release_notes/RELEASE.txt [hunk 3]
```diff
@@ -38,6 +38,33 @@ New features and changes
...
+
+        CMake:        HDF_ENABLE_NETCDF_TOOLS
+        Autotools:    --enable-netcdf-tools
+
+      This option can be set independently of the option that controls netCDF
+      name mangling (HDF_ENABLE_NETCDF / --enable-netcdf) as no C library
+      functionality is affected by that option.
+
+      This option is enabled by default. HDF4_BUILD_TOOLS must be enabled to
+      build the netCDF tools.
```
Comment (lrknox) [inline release_notes/RELEASE.txt:64]: HDF4_BUILD_TOOLS is required for CMake.  Is there a similar requirement for autotools, or is it automatic?

**Pull request merged**